src/models/collaboration/usePatchCollaboration.js

```diff
@@ -1,48 +1,45 @@
-import { useState } from 'react';
 import axios from 'axios';
-import { get } from 'lodash-es';
-import { formatError } from '../../utils/formatters';
+import { useMutation } from 'react-query';
 
 export default function usePatchCollaboration() {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState(null);
-  const [success, setSuccess] = useState(false);
-
-  const patchCollaboration = async (collaborationId, data) => {
-    try {
-      setLoading(true);
-      const patchResponse = await axios({
-        url: `${__houston_url__}/api/v1/collaborations/${collaborationId}`,
-        withCredentials: true,
-        method: 'patch',
-        data,
-      });
-      const responseStatus = get(patchResponse, 'status');
-      const successful = responseStatus === 200;
-      if (successful) {
-        setLoading(false);
-        setSuccess(true);
-        setError(null);
-        return true;
-      }
-
-      setError(formatError(patchResponse));
-      setSuccess(false);
-      return false;
-    } catch (postError) {
-      setLoading(false);
-      setError(formatError(postError));
-      setSuccess(false);
-      return false;
+  const collabPatchMutation = useMutation(async dataObj => {
+    const collabId = dataObj?.collabId;
+    if (dataObj?.dataReverse) {
+      // it's two-way
+      return axios.all([
```

Comment:
isLoading works fine now with Promise.all.

I find myself mildly curious with how isLoading would work in terms of your second suggestion:
```
const response1 = await axios({ ... })
const response2 = await axios({ ... 
``` 

Do you mean that we wouldn't have used react-query at all in this latter case? Feel free to ignore since this is a bridge that we did not have to cross.

Reply:
this second route would have been inside of react-queryland too
```
useMutation(async dataObj => {
  const response1 = await axios({ ... });
  const response2 = await axios({ ... });
  const success1 = some test on response1;
  const success2 = some test on response2;
  return success1 && success2;
})
```